src/metabase/models/setting.clj

```diff
@@ -1093,16 +1113,15 @@
   are returned."
   [& {:as options}]
   ;; ignore Database-local values, but not User-local values
-  (binding [*database-local-values* nil]
-    (into
-     []
-     (comp (filter (fn [setting]
-                     (and (not= (:visibility setting) :internal)
-                          (or api/*is-superuser?*
-                              (not= (:visibility setting) :admin))
-                          (not= (:database-local setting) :only))))
-           (map #(m/mapply user-facing-info % options)))
-     (sort-by :name (vals @registered-settings)))))
+  (let [writable-visibilities (current-user-writable-visibilities)]
+    (binding [*database-local-values* nil]
+      (into
+       []
+       (comp (filter (fn [setting]
+                       (and (contains? writable-visibilities (:visibility setting))
+                            (not= (:database-local setting) :only))))
+             (map #(m/mapply user-facing-info % options)))
+       (sort-by :name (vals @registered-settings))))))
```

Comment:
This is doing a slightly tighter permissions check here. Previously, an implicit assumption of`admin-writable-settings` is that it can only be called in contexts where the user is either a superuser, or has advanced setting access.

Before the change, if you called this function when the user doesn't have advanced setting access would incorrectly return all the settings that don't have :admin or :internal visibility. However, non-admins without advanced setting access shouldn't have write access to any settings at all. It should return an empty list. This change fixes that.

Reply:
Makes sense, nice catch